k8sdeps/transformer/patch/conflictdetector.go

```diff
@@ -123,19 +124,25 @@ func (smp *strategicMergePatch) findConflict(
 }
 
 func (smp *strategicMergePatch) mergePatches(patch1, patch2 *resource.Resource) (*resource.Resource, error) {
+	if hasDeleteDirectiveMarker(patch2.Map()) {
+		if hasDeleteDirectiveMarker(patch1.Map()) {
+			return nil, fmt.Errorf("cannot merge patches both containing '$patch: delete' directives")
+		}
+		patch1, patch2 = patch2, patch1
+	}
 	mergeJSONMap, err := strategicpatch.MergeStrategicMergeMapPatchUsingLookupPatchMeta(
 		smp.lookupPatchMeta, patch1.Map(), patch2.Map())
 	return smp.rf.FromMap(mergeJSONMap), err
 }
 
-// mergePatches merge and index patches by OrgId.
+// MergePatches merge and index patches by OrgId.
 // It errors out if there is conflict between patches.
 func MergePatches(patches []*resource.Resource,
 	rf *resource.Factory) (resmap.ResMap, error) {
 	rc := resmap.New()
 	for ix, patch := range patches {
 		id := patch.OrgId()
-		existing := rc.GetMatchingResourcesByOriginalId(id.Equals)
+		existing := rc.GetMatchingResourcesByOriginalId(id.GvknEquals)
```

Comment:
We just changed it from GvknEquals to Equals because it fails in multi name space environment.

Reply:
Fixed as well.